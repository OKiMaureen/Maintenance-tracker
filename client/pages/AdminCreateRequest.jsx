import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Validator from 'validatorjs';
import { bindActionCreators } from 'redux';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import createRequestAction from '../actions/createRequestAction';
import Header from '../components/Header';
import clearMessageAction from '../actions/clearMeassageAction';
import logoutAction from '../actions/logoutAction';
import Request from '../components/Request';

export class CreateRequest extends Component {
    /**
     * @constructor
     * @param {*} props
     */
    state = {
      department: '',
      equipment: '',
      serialnumber: '11111111',
      title: '',
      description: '',
      errors: {},
    }
    validate = () => {
      const validations = new Validator(
        {
          department: this.state.department,
          equipment: this.state.equipment,
          serialnumber: this.state.serialnumber,
          title: this.state.title,
          description: this.state.description,
        },

        {
          title: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:5', 'max:20'],
          department: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i'],
          equipment: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i'],
          serialnumber: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:8', 'max:8'],
          description: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:3', 'max:50'],
        },
        {
          'min.title': 'The :attribute must not be less than 5 characters.',
          'max.title': 'The :attribute must not be greater than 20 characters.',
          'min.serialnumber': 'The :attribute must be only 8 characters.',
          'max.serialnumber': 'The :attribute must be only 8 characters.',
          'min.description': 'The :attribute must not be less than 3 characters.',
          'max.description': 'The :attribute must not be greater than 50 characters.',
        },
      );
      if (validations.fails()) {
        const errors = validations.errors.all();
        this.setState({
          errors,
        });

        return false;
      }

      return true;
    }

    handleChange = (e) => {
      if (this.validate()) {
        this.setState({
          errors: {
            ...this.state.errors,
            [e.target.name]: '',
          },
        });
      }
      this.setState({ [e.target.name]: e.target.value });
      const { clearMessage, requestDetail } = this.props;
      if (requestDetail.error) clearMessage();
      return true;
    }
    handleSubmit = (event) => {
      event.preventDefault();
      if (this.validate()) {
        this.props.createRequestAction(this.state, this.props.history);
      }
      return true;
    }
    logout = () => {
      const { history, logout } = this.props;
      logout();
      localStorage.clear();
      history.push('/');
      return true;
    }
    render() {
      const { requestDetail, userDetail } = this.props;
      const {
        title, description, equipment, errors, department,
      } = this.state;
      return (
        <div>
          <Header><Link to="/">Home</Link><Link to="/admincreaterequest">Create request</Link> <Link to="/adminrequests">All requests</Link><li><button className="header-btn" onClick={this.logout}>Logout</button></li></Header>
          <Request
            handleChange={this.handleChange}
            handleSubmit={this.handleSubmit}
            title={title}
            description={description}
            equipment={equipment}
            requestDetail={requestDetail}
            userDetail={userDetail}
            errors={errors}
            department={department}
            titleText="Create Request"
            buttonText="Create Request"
          />
        </div>
      );
    }
}

const mapStateToProps = state => ({
  requestDetail: state.createRequest,
  userDetail: state.authUser,
});


const mapDispatchToProps = dispatch => bindActionCreators({
  createRequestAction,
  clearMessage: clearMessageAction,
  logout: logoutAction,
}, dispatch);

CreateRequest.propTypes = {
  requestDetail: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  userDetail: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  createRequestAction: PropTypes.func.isRequired,
  clearMessage: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};


export default connect(mapStateToProps, mapDispatchToProps)(CreateRequest);

