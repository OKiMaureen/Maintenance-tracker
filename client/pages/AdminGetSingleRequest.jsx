import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import i from 'short-id';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import adminGetSingleRequestAction, { updateReq, disapproveRequest, resolveRequest } from '../actions/adminSingleRequestAction';
import logoutAction from '../actions/logoutAction';
import Header from '../components/Header';
import Footer from '../components/Footer';

export class AdminSingleRequest extends Component {
  componentDidMount() {
    const { adminSingleRequest, match } = this.props;
    adminSingleRequest(match.params.id);
  }
  logout = () => {
    const { history, logout } = this.props;
    logout();
    localStorage.clear();
    history.push('/');
    return true;
  }
  updateRequest = (e) => {
    const status = e.target.id;
    const { update, oneRequest } = this.props;
    update(oneRequest.id, status);
    return true;
  }

  render() {
    const {
      oneRequest,
    } = this.props;
    return (
      <div>
        <Header>
          <Link to="/">Home</Link>
          <Link to="/createrequest">Create Request</Link>
          <Link to="/adminrequests">All requests</Link>
          <li><button className="header-btn" onClick={this.logout}>Logout</button></li>
        </Header>

        <section className="allrequests">
          {oneRequest ?
            <div className="detailsrequest-card" key={i.generate()}>
              <div className="status">
                {oneRequest.requeststatus === 'approved' || oneRequest.requeststatus === 'resolved' ?
                  <button id="approve" className="disabled" disabled onClick={this.updateRequest}>Approve</button> :
                  <button className="green" id="approve" onClick={this.updateRequest}>Approve</button>}
                {oneRequest.requeststatus === 'disapproved' || oneRequest.requeststatus === 'resolved' ?
                  <button id="disapprove" className="disabled" disabled onClick={this.updateRequest}>Disapprove</button> :
                  <button className="red" id="disapprove" onClick={this.updateRequest}>Disapprove</button>}
                {oneRequest.requeststatus === 'disapproved' || oneRequest.requeststatus === 'resolved' || oneRequest.requeststatus === 'pending' ?
                  <button id="resolved" className="disabled" disabled onClick={this.updateRequest}>Resolve</button> :
                  <button className="green" id="resolve" onClick={this.updateRequest}>Resolve</button>}
              </div>
              <p>
                <label>Title:</label>{oneRequest.title}
              </p>
              <p>
                <label>Department:</label>{oneRequest.department}
              </p>
              <p>
                <label>Equipment: </label>{oneRequest.equipment}
              </p>
              <p>
                <label>Description: </label>{oneRequest.description}
              </p>
              <div className="container">
                <div className="status fixed">
                  {oneRequest.requeststatus === 'pending' ? <label className="yellow">{oneRequest.requeststatus}</label> : ''}
                  {oneRequest.requeststatus === 'approved' ? <label className="green">{oneRequest.requeststatus}</label> : ''}
                  {oneRequest.requeststatus === 'disapproved' ? <label className="red">{oneRequest.requeststatus}</label> : ''}
                  {oneRequest.requeststatus === 'resolved' ? <label className="green">{oneRequest.requeststatus}</label> : ''}
                </div>
              </div>
            </div> : '' }

        </section>
        <Footer />
      </div>

    );
  }
}

const mapStateToProps = state => ({
  oneRequest: state.adminSingleRequest.request,
  userDetail: state.authUser,
  logout: logoutAction,

});
const mapDispatchToProps = dispatch => bindActionCreators({
  adminSingleRequest: adminGetSingleRequestAction,
  update: updateReq,
  disapprove: disapproveRequest,
  resolve: resolveRequest,

}, dispatch);

AdminSingleRequest.propTypes = {
  oneRequest: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  adminSingleRequest: PropTypes.func.isRequired,
  update: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  match: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};
export default connect(mapStateToProps, mapDispatchToProps)(AdminSingleRequest);
