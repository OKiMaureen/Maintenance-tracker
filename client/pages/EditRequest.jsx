import PropTypes from 'prop-types';
import Validator from 'validatorjs';
import { bindActionCreators } from 'redux';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import editRequestAction from '../actions/editRequestAction';
import clearMessageAction from '../actions/clearMeassageAction';
import logoutAction from '../actions/logoutAction';
import Request from '../components/Request';

export class EditRequest extends Component {
  constructor(props) {
    super(props);
    const { request } = props;
    this.state = {
      department: request.department || '',
      equipment: request.equipment || '',
      serialnumber: '11111111',
      title: request.title || '',
      description: request.description || '',
      errors: {},
    };
  }
  validate = () => {
    const validations = new Validator(
      {
        department: this.state.department,
        equipment: this.state.equipment,
        serialnumber: this.state.serialnumber,
        title: this.state.title,
        description: this.state.description,
      },

      {
        title: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:5', 'max:20'],
        department: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i'],
        equipment: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i'],
        serialnumber: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:8', 'max:8'],
        description: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:3', 'max:50'],
      },
      {
        'min.title': 'The :attribute must not be less than 5 characters.',
        'max.title': 'The :attribute must not be greater than 20 characters.',
        'min.serialnumber': 'The :attribute must be only 8 characters.',
        'max.serialnumber': 'The :attribute must be only 8 characters.',
        'min.description': 'The :attribute must not be less than 3 characters.',
        'max.description': 'The :attribute must not be greater than 50 characters.',
      },
    );
    if (validations.fails()) {
      const errors = validations.errors.all();
      this.setState({
        errors,
      });

      return false;
    }

    return true;
  }

    handleChange = (e) => {
      if (this.validate()) {
        this.setState({
          errors: {
            ...this.state.errors,
            [e.target.name]: '',
          },
        });
      }
      this.setState({ [e.target.name]: e.target.value });
      const { clearMessage, requestDetail } = this.props;
      if (requestDetail.error) clearMessage();
      return true;
    }
    handleSubmit = (event) => {
      const { match, history, updateRequest } = this.props;
      event.preventDefault();
      if (this.validate()) {
        updateRequest(this.state, match.params.id, history);
      }
      return true;
    }
    logout = () => {
      const { history, logout } = this.props;
      logout();
      localStorage.clear();
      history.push('/');
      return true;
    }
    render() {
      const { requestDetail, userDetail, editRequest } = this.props;
      const {
        title, description, equipment, errors, department,
      } = this.state;
      return (
        <Request
          handleChange={this.handleChange}
          handleSubmit={this.handleSubmit}
          title={title}
          description={description}
          equipment={equipment}
          logout={this.logout}
          editRequest={editRequest}
          requestDetail={requestDetail}
          userDetail={userDetail}
          errors={errors}
          department={department}
          titleText="Edit Article"
          buttonText="Submit"
        />
      );
    }
}

const mapStateToProps = state => ({
  request: state.singleRequest.request,
  requestDetail: state.createRequest,
  userDetail: state.authUser,
  editRequest: state.editRequest,
});


const mapDispatchToProps = dispatch => bindActionCreators({
  clearMessage: clearMessageAction,
  logout: logoutAction,
  updateRequest: editRequestAction,
}, dispatch);

EditRequest.propTypes = {
  request: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  requestDetail: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  userDetail: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  editRequest: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  updateRequest: PropTypes.func.isRequired,
  clearMessage: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
  match: PropTypes.func.isRequired,
};


export default connect(mapStateToProps, mapDispatchToProps)(EditRequest);

