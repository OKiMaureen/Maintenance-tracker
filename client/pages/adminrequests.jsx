import React from 'react';

const AdminRequests = () => ((
    <div>
        <section className="approverequests">
        <div className="approverequests-card">
          <p>
            <label>Title:</label>Repair Computer</p>
          <p>
            <label>S/N: </label>HP000001</p>
          <p>
          </p><div className="status">
            
          </div>
        </div>
        <div className="approverequests-card">
          <p>
            <label>Title:</label>Repair Computer</p>
          <p>
            <label>S/N: </label>HP000001</p>
          <p>
          </p><div className="status">
         </div>
        </div>
        <div className="approverequests-card">
          <p>
            <label>Title:</label>Repair Computer</p>
          <p>
            <label>S/N: </label>HP000001</p>
          <p>
          </p><div className="status">
          </div>
        </div>
        <div className="approverequests-card">
          <p>
            <label>Title:</label>Repair Computer</p>
          <p>
            <label>S/N: </label>HP000001</p>
          <p>
          </p><div className="status">
           </div>
        </div>
        <div className="approverequests-card">
          <p>
            <label>Title:</label>Repair Computer</p>
          <p>
            <label>S/N: </label>HP000001</p>
          <p>
          </p><div className="status">
          </div>
        </div>
      </section>
    </div>
));
export default AdminRequests;