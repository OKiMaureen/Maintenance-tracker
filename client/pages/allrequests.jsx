import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import id from 'short-id';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import allRequestAction from '../actions/allRequestAction';
import logoutAction from '../actions/logoutAction';
import Header from '../components/Header';
import Footer from '../components/Footer';

export class AllRequests extends Component {
  componentDidMount() {
    const { allRequest } = this.props;
    allRequest();
  }
  logout = () => {
    const { history, logout } = this.props;
    if (logout()) {
      localStorage.clear();
      history.push('/');
    }
    return true;
  }
  render() {
    const { requests = [] } = this.props;
    return (
      <div>
        <Header>
          <Link to="/">Home</Link>
          <Link to="/createrequest">Create Request</Link>
          <Link to="/allrequests" className="current">My requests</Link>
          <button className="header-btn" onClick={this.logout}>Logout</button>
        </Header>

        <section className="allrequests">
          {requests && requests.length !== 0 ? requests.map(request =>

          (
            <div className="requests-card" key={id.generate}>
              <p>
                <label>Title:</label>{request.title}
              </p>
              <p>
                <label>Equipment: </label>{request.equipment}
              </p>
              <div className="container">
                <div className="status fixed">
                  {request.requeststatus === 'pending' ? <label className="yellow">{request.requeststatus}</label> : ''}
                  {request.requeststatus === 'approved' ? <label className="green">{request.requeststatus}</label> : ''}
                  {request.requeststatus === 'disapproved' ? <label className="red">{request.requeststatus}</label> : ''}
                  {request.requeststatus === 'resolved' ? <label className="green">{request.requeststatus}</label> : ''}
                </div>
                <div className="status flex-item ">
                  <p>
                    { <Link to={`/singlerequest/${request.id}`}> More</Link>}

                  </p>
                </div>
              </div>
            </div>)) :
          <div className="requests-card no-request" >
            <p className="message-centered">
              You do not have any requests yet!!! Go ahead and create a request.
            </p>
          </div>
          }

        </section>
        <Footer />
      </div>

    );
  }
}

const mapStateToProps = state => ({
  requests: state.requests.requests,
  error: state.requests.error,
  userDetail: state.authUser,

});


const mapDispatchToProps = dispatch => bindActionCreators({
  allRequest: allRequestAction,
  logout: logoutAction,
}, dispatch);

AllRequests.propTypes = {
  requests: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  allRequest: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};
export default connect(mapStateToProps, mapDispatchToProps)(AllRequests);
