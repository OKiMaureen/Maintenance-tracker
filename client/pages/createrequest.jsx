import PropTypes from 'prop-types';
import Validator from 'validatorjs';
import { Link } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import createRequestAction from '../actions/createRequestAction';
import loader from '../assets/images/spinloader.gif';
import Header from '../components/Header';
import Footer from '../components/Footer';

export class CreateRequest extends Component {
    /**
     * @constructor
     * @param {*} props
     */
    state = {
      department: '',
      equipment: '',
      serialnumber: '11111111',
      title: '',
      description: '',
      errors: {},
    }
    validate = () => {
      const validations = new Validator(
        {
          department: this.state.department,
          equipment: this.state.equipment,
          serialnumber: this.state.serialnumber,
          title: this.state.title,
          description: this.state.description,
        },

        {
          title: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:5', 'max:20'],
          department: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i'],
          equipment: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i'],
          serialnumber: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:8', 'max:8'],
          description: ['required', 'string', 'regex:/^[a-z\\d\\-_,.*()!\\s]+$/i', 'min:3', 'max:50'],
        },
        {
          'min.title': 'The :attribute must not be less than 5 characters.',
          'max.title': 'The :attribute must not be greater than 20 characters.',
          'min.serialnumber': 'The :attribute must be only 8 characters.',
          'max.serialnumber': 'The :attribute must be only 8 characters.',
          'min.description': 'The :attribute must not be less than 3 characters.',
          'max.description': 'The :attribute must not be greater than 50 characters.',
        },
      );
      if (validations.fails()) {
        const errors = validations.errors.all();
        this.setState({
          errors,
        });

        return false;
      }

      return true;
    }

    handleChange = (e) => {
      if (this.validate()) {
        this.setState({
          errors: {
            ...this.state.errors,
            [e.target.name]: '',
          },
        });
      }
      this.setState({ [e.target.name]: e.target.value });
      return true;
    }
    handleSubmit = (event) => {
      event.preventDefault();
      if (this.validate()) {
        this.props.createRequestAction(this.state, this.props.history);
      }
      return true;
    }
    logout = () => {
      localStorage.clear();
      this.props.history.push('/');
      return true;
    }
    render() {
      return (
        <div>
          <Header><Link to="/">Home</Link> <Link to="/allrequests">My request</Link><button className="header-btn" onClick={this.logout}>Logout</button></Header>
          <section className="request">
            <form className="request-form">
              <h3>Create Request</h3>
              <br />

              <p>Fill the form below to indicate equipments in need of repair or maintenance</p>
              <br />
              {
                this.props.requestDetail.error ? <span className="validation-error">{this.props.requestDetail.error}</span> : ''
              }
              {this.props.userDetail.checkStatus.isLoading ? (<span className="loader"><img src={loader} alt="loader" /></span>) : '' }
              <input type="text" name="department" id="department" placeholder="Department" onChange={this.handleChange} />
              {
                this.state.errors.department ? this.state.errors.department.map(error => <span className="validation-error" key={error}>{error}</span>) : ''
              }
              <p className="message" id="departmentE" />
              <br />
              <input type="text" name="equipment" id="equipment" placeholder="Equipment" onChange={this.handleChange} />
              {
                  this.state.errors.equipment ? this.state.errors.equipment.map(error => <span className="validation-error" key={error}>{error}</span>) : ''
                }
              <p className="message" id="equipmentE" />
              <br />
              <input type="text" name="title" id="requestTitle" placeholder="Request Title" onChange={this.handleChange} />
              {
              this.state.errors.title ? this.state.errors.title.map(error => <span className="validation-error" key={error}>{error}</span>) : ''
              }
              <p className="message" id="titleE" />
              <br />
              <textarea name="description" id="requestDescription" cols="20" rows="5" placeholder="Request Description" onChange={this.handleChange} />
              {
                  this.state.errors.description ? this.state.errors.description.map(error => <span className="validation-error" key={error}>{error}</span>) : ''
                }
              <p className="message" id="descriptionE" />
              <br />
              <button type="submit" className="button" id="createRequest" onClick={this.handleSubmit}>Create Request</button>
            </form>
          </section>
          <Footer />
        </div>

      );
    }
}

const mapStateToProps = state => ({
  requestDetail: state.createRequest,
  userDetail: state.authUser,
});


const mapDispatchToProps = dispatch => bindActionCreators({
  createRequestAction,
}, dispatch);

CreateRequest.propTypes = {
  requestDetail: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  userDetail: PropTypes.shape({
    checkStatus: PropTypes.object,
    error: PropTypes.string,
  }).isRequired,
  createRequestAction: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.object).isRequired,
};


export default connect(mapStateToProps, mapDispatchToProps)(CreateRequest);

