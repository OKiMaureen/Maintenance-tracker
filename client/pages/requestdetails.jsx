import React from 'react';

const UserDetails = () => ((
  <div>
    <section className="details-requests">
    
        <div className="detailsrequest-card">
          <div className="status">
            <button className="green">Approve</button>
            <button className="red">Disapprove</button>
            <button className="green">Resolve</button>
          </div>
          <p>
            <label>Title:</label>Repair Computer
          </p>
          <br />
          <p><label>Department:</label>Technical</p>
          <br />
          <p><label>Equipment: </label>HP computer</p>
          <br />
          <p><label>S/N: </label>HP000001</p>
          <br />
          <p><label>Description: </label>Battery is faulty, and It fell down, and sice I fixed it, I have not been able to use it.
            Battery is faulty, and It fell down, and sice I fixed it, I have not been able to use it.
          </p>
          <br />
          <div className="status">
            <label className="green">Resolved</label>
          </div>
        </div>
     

    </section>

  </div>
));
export default UserDetails;
