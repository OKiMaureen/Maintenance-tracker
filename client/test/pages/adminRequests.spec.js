import React from 'react';
import { shallow } from 'enzyme';
import { AdminRequests } from '../../pages/adminrequests';


describe('<Admin Requests />', () => {
  let wrapper;
  let props = {
    adminRequests: () => {},
    logout: () => {},
    requests: [
      {
        id: 1,
        user_id: 1,
        title: 'repair computer',
        department: 'technology',
        equipment: 'computer',
        requeststatus: 'pending',
        description: 'repair laptop',
        serialnumber: 111111111,
      },
      {
        id: 2,
        user_id: 2,
        title: 'repair computer',
        department: 'technology',
        equipment: 'computer',
        requeststatus: 'pending',
        description: 'repair laptop',
        serialnumber: 111111111,
      },
    ],
    history: {
      push: jest.fn(),
    },
  };

  it('stores a snapshot of the component', () => {
    wrapper = shallow(<AdminRequests {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it('Should not display request card when request does not exists', () => {
    props = {
      adminRequests: () => {},
      requests: [],
    };

    wrapper = shallow(<AdminRequests {...props} />);
    expect(wrapper.find('label').exists()).toBe(false);
  });
  it('Should handle logout method', () => {
    props = {
      adminRequests: () => {},
      logout: () => {},
      history: {
        push: jest.fn(),
      },
    };
    wrapper = shallow(<AdminRequests {...props} />);
    expect(wrapper.instance().logout()).toBe(true);
  });
});
