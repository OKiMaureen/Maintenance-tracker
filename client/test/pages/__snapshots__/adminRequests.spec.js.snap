// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Admin Requests /> stores a snapshot of the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminRequests
    adminRequests={[Function]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    logout={[Function]}
    requests={
      Array [
        Object {
          "department": "technology",
          "description": "repair laptop",
          "equipment": "computer",
          "id": 1,
          "requeststatus": "pending",
          "serialnumber": 111111111,
          "title": "repair computer",
          "user_id": 1,
        },
        Object {
          "department": "technology",
          "description": "repair laptop",
          "equipment": "computer",
          "id": 2,
          "requeststatus": "pending",
          "serialnumber": 111111111,
          "title": "repair computer",
          "user_id": 2,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header>
          <Link
            replace={false}
            to="/"
          >
            Home
          </Link>
          <Link
            replace={false}
            to="admincreaterequest"
          >
            Create Request
          </Link>
          <Link
            className="current"
            replace={false}
            to="adminrequests"
          >
            All Requests
          </Link>
          <button
            className="header-btn"
            onClick={[Function]}
          >
            Logout
          </button>
        </Header>,
        <AllRequestsComponent
          linkRoute="/adminrequests"
          linkText="All Requests"
          logout={[Function]}
          requests={
            Array [
              Object {
                "department": "technology",
                "description": "repair laptop",
                "equipment": "computer",
                "id": 1,
                "requeststatus": "pending",
                "serialnumber": 111111111,
                "title": "repair computer",
                "user_id": 1,
              },
              Object {
                "department": "technology",
                "description": "repair laptop",
                "equipment": "computer",
                "id": 2,
                "requeststatus": "pending",
                "serialnumber": 111111111,
                "title": "repair computer",
                "user_id": 2,
              },
            ]
          }
          route="/requestdetails/"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/"
            >
              Home
            </Link>,
            <Link
              replace={false}
              to="admincreaterequest"
            >
              Create Request
            </Link>,
            <Link
              className="current"
              replace={false}
              to="adminrequests"
            >
              All Requests
            </Link>,
            <button
              className="header-btn"
              onClick={[Function]}
            >
              Logout
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Home",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Create Request",
              "replace": false,
              "to": "admincreaterequest",
            },
            "ref": null,
            "rendered": "Create Request",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "All Requests",
              "className": "current",
              "replace": false,
              "to": "adminrequests",
            },
            "ref": null,
            "rendered": "All Requests",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Logout",
              "className": "header-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Logout",
            "type": "button",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "linkRoute": "/adminrequests",
          "linkText": "All Requests",
          "logout": [Function],
          "requests": Array [
            Object {
              "department": "technology",
              "description": "repair laptop",
              "equipment": "computer",
              "id": 1,
              "requeststatus": "pending",
              "serialnumber": 111111111,
              "title": "repair computer",
              "user_id": 1,
            },
            Object {
              "department": "technology",
              "description": "repair laptop",
              "equipment": "computer",
              "id": 2,
              "requeststatus": "pending",
              "serialnumber": 111111111,
              "title": "repair computer",
              "user_id": 2,
            },
          ],
          "route": "/requestdetails/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header>
            <Link
              replace={false}
              to="/"
            >
              Home
            </Link>
            <Link
              replace={false}
              to="admincreaterequest"
            >
              Create Request
            </Link>
            <Link
              className="current"
              replace={false}
              to="adminrequests"
            >
              All Requests
            </Link>
            <button
              className="header-btn"
              onClick={[Function]}
            >
              Logout
            </button>
          </Header>,
          <AllRequestsComponent
            linkRoute="/adminrequests"
            linkText="All Requests"
            logout={[Function]}
            requests={
              Array [
                Object {
                  "department": "technology",
                  "description": "repair laptop",
                  "equipment": "computer",
                  "id": 1,
                  "requeststatus": "pending",
                  "serialnumber": 111111111,
                  "title": "repair computer",
                  "user_id": 1,
                },
                Object {
                  "department": "technology",
                  "description": "repair laptop",
                  "equipment": "computer",
                  "id": 2,
                  "requeststatus": "pending",
                  "serialnumber": 111111111,
                  "title": "repair computer",
                  "user_id": 2,
                },
              ]
            }
            route="/requestdetails/"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/"
              >
                Home
              </Link>,
              <Link
                replace={false}
                to="admincreaterequest"
              >
                Create Request
              </Link>,
              <Link
                className="current"
                replace={false}
                to="adminrequests"
              >
                All Requests
              </Link>,
              <button
                className="header-btn"
                onClick={[Function]}
              >
                Logout
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Home",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Create Request",
                "replace": false,
                "to": "admincreaterequest",
              },
              "ref": null,
              "rendered": "Create Request",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "All Requests",
                "className": "current",
                "replace": false,
                "to": "adminrequests",
              },
              "ref": null,
              "rendered": "All Requests",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Logout",
                "className": "header-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Logout",
              "type": "button",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "linkRoute": "/adminrequests",
            "linkText": "All Requests",
            "logout": [Function],
            "requests": Array [
              Object {
                "department": "technology",
                "description": "repair laptop",
                "equipment": "computer",
                "id": 1,
                "requeststatus": "pending",
                "serialnumber": 111111111,
                "title": "repair computer",
                "user_id": 1,
              },
              Object {
                "department": "technology",
                "description": "repair laptop",
                "equipment": "computer",
                "id": 2,
                "requeststatus": "pending",
                "serialnumber": 111111111,
                "title": "repair computer",
                "user_id": 2,
              },
            ],
            "route": "/requestdetails/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
